server/src/test/java/io/crate/integrationtests/WindowFunctionsIntegrationTest.java

```diff
@@ -288,4 +289,11 @@ public void test_select_with_standalone_ref_and_subquery_filter_in_window_functi
         assertThat(printedTable(response.rows()), is("1| [1]\n" +
                                                      "2| [1]\n"));
     }
+
+    @Test
+    public void test_can_select_only_one_field_if_subquery_selects_this_field_and_window_func_over_another_field() {
```

Comment:
Now that we'll change the plan by removing the WindowAgg, I think we can easily turn this into a planner test case?

Reply:
I'm currently trying to test 2 window functions case. I mean having 2 different partitions, one is used in top select (so, we don't remove WindowAgg) and another is not used - so we can again catch the bug as WindowAgg is not removed but this time on third field z from query:

`create table t (x int, y int, z int)`
```
select x, ry from 
    (select 
       x, 
       ROW_NUMBER() OVER (PARTITION BY y) as ry,
       ROW_NUMBER() OVER (PARTITION BY z) as rz  
    from t
    ) t1
```
but it passes - either scenario is somehow invalid or I'm testing it wrong.

Reply:
done in https://github.com/crate/crate/pull/12116/commits/5aebeadf172ebd1ed90a29405217c93e04218ec2

looks like scenario I posted above never happens - in debugger `newWindowFunctions` was empty - I didn't expect that as one window func result was used in upper select but I probably missed some other factor.

Reply:
Also if the partition column differs, each window function will result in a dedicated window agg projection.